include/boost/di.hpp

```diff
@@ -43,6 +43,9 @@ BOOST_DI_CFG_FWD
   {}
 #elif defined(_MSC_VER)
 #define __MSVC__
+#if !defined(__has_extension)
+#define __has_extension(...) 1
```

Comment:
`__has_extension` is not a thing in msvc. However it does support `__is_constructable`.
We can even employ `std::is_constructable` if defined